src/storage/storage_structure/disk_overflow_file.cpp

```diff
@@ -39,45 +47,24 @@ void DiskOverflowFile::readStringToVector(
     bufferManager.unpin(*fileHandleToPin, pageIdxToPin);
 }
 
-void DiskOverflowFile::scanSequentialStringOverflow(TransactionType trxType, ValueVector& vector) {
-    FileHandle* cachedFileHandle = nullptr;
-    page_idx_t cachedPageIdx = UINT32_MAX;
-    uint8_t* cachedFrame = nullptr;
-    for (auto i = 0u; i < vector.state->selVector->selectedSize; ++i) {
-        auto pos = vector.state->selVector->selectedPositions[i];
-        if (vector.isNull(pos)) {
-            continue;
-        }
-        auto& kuString = ((ku_string_t*)vector.getData())[pos];
-        if (ku_string_t::isShortString(kuString.len)) {
-            continue;
-        }
-        page_idx_t pageIdx = UINT32_MAX;
-        uint16_t pagePos = UINT16_MAX;
-        TypeUtils::decodeOverflowPtr(kuString.overflowPtr, pageIdx, pagePos);
-        auto [fileHandleToPin, pageIdxToPin] =
-            StorageStructureUtils::getFileHandleAndPhysicalPageIdxToPin(
-                fileHandle, pageIdx, *wal, trxType);
-        if (pageIdxToPin == cachedPageIdx) { // cache hit
-            InMemOverflowBufferUtils::copyString(
-                (char*)(cachedFrame + pagePos), kuString.len, kuString, vector.getOverflowBuffer());
-            continue;
-        }
-        // cache miss
-        if (cachedPageIdx != UINT32_MAX) { // unpin cached frame
-            bufferManager.unpin(*cachedFileHandle, cachedPageIdx);
-        }
-        // pin new frame and update cache
-        auto frame = bufferManager.pin(*fileHandleToPin, pageIdxToPin);
-        InMemOverflowBufferUtils::copyString(
-            (char*)(frame + pagePos), kuString.len, kuString, vector.getOverflowBuffer());
-        cachedFileHandle = fileHandleToPin;
-        cachedPageIdx = pageIdxToPin;
-        cachedFrame = frame;
+void DiskOverflowFile::lookupString(TransactionType trxType, ku_string_t& kuStr,
+    InMemOverflowBuffer& inMemOverflowBuffer, OverflowPageCache& overflowPageCache) {
+    if (ku_string_t::isShortString(kuStr.len)) {
+        return;
     }
-    if (cachedPageIdx != UINT32_MAX) {
-        bufferManager.unpin(*cachedFileHandle, cachedPageIdx);
+    PageByteCursor cursor;
+    TypeUtils::decodeOverflowPtr(kuStr.overflowPtr, cursor.pageIdx, cursor.offsetInPage);
+    auto [fileHandleToPin, pageIdxToPin] =
+        StorageStructureUtils::getFileHandleAndPhysicalPageIdxToPin(
+            fileHandle, cursor.pageIdx, *wal, trxType);
+    if (pageIdxToPin != overflowPageCache.pageIdx) { // cache miss
+        unpinOverflowPageCache(overflowPageCache);
+        overflowPageCache.frame = bufferManager.pin(*fileHandleToPin, pageIdxToPin);
+        overflowPageCache.fileHandle = fileHandleToPin;
+        overflowPageCache.pageIdx = pageIdxToPin;
```

Comment:
I would consider wrapping these three lines as `pinOverflowPageCache`, but i don't have a strong opinion on this, up to you.
```
        overflowPageCache.frame = bufferManager.pin(*fileHandleToPin, pageIdxToPin);
        overflowPageCache.fileHandle = fileHandleToPin;
        overflowPageCache.pageIdx = pageIdxToPin;
```

Reply:
I think it makes sense since I already have `unpinOverflowPageCache`, which definitely increases readability. thanks